packages/runtime/container-runtime/src/batchTracker.ts

```diff
@@ -0,0 +1,82 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ITelemetryLogger } from "@fluidframework/common-definitions";
+import { assert } from "@fluidframework/common-utils";
+import { ISequencedDocumentMessage } from "@fluidframework/protocol-definitions";
+import { ChildLogger } from "@fluidframework/telemetry-utils";
+import EventEmitter from "events";
+
+export class BatchTracker {
+    private readonly logger: ITelemetryLogger;
+    private startBatchSequenceNumber: number | undefined;
+    private trackedBatchCount: number = 0;
+    private batchProcessingStartTimeStamp: number | undefined;
+
+    constructor(
+        private readonly batchEventEmitter: EventEmitter,
+        logger: ITelemetryLogger,
+        private readonly opCountThreshold: number,
+        private readonly batchCountSamplingRate: number,
+        private readonly dateTimeProvider: () => number = () => Date.now(),
```

Comment:
I think performance.now() (the one in common package) is preferred for relative measurements, but I do not remember why. It sued to have higher res, and it definitely does not go back in time (do not remember about Date.now - I think it can)

Reply:
I can switch it. I was thinking the resolution here (microseconds) doesn't really matter in this context.